storage/storage_image.go

```diff
@@ -360,6 +367,57 @@ func (s *storageImageDestination) HasThreadSafePutBlob() bool {
 	return true
 }
 
+// SupportsBlobLocks indicates whether the ImageDestination supports blob
+// locking.
+func (s *storageImageDestination) SupportsBlobLocks() bool {
+	return true
+}
+
+// LockBlob can be used to synchronize operations on it (e.g., copying).
+// Note that any unlocked lock will be unlocked in Close().
+func (s *storageImageDestination) LockBlob(b types.BlobInfo) error {
+	// anonymous function to not worry about errors paths and panics
+	getLocker := func() (storage.Locker, error) {
```

Comment:
(_Absolutely_ non-blocking:

1. consider
    ```go
    locker, err := func() (storage.Locker, error) {
       …
    }()
    ```
    I am honestly not sure whether it is more readable — eliminating the one-use identifier is a plus, OTOH one needs to read up to the end of the function definition to see what the code does with the lambda, in either case.
2. ”to not worry about error paths“ is a bit unclear about what the worry is exactly; if you mean “to ensure `s.digestLockMutex` is unlocked on all paths”, say so.

)

Reply:
Nice idea! Done.